### PR TITLE
git-filter-repo: fix broken --help option

### PR DESCRIPTION
**Pull request by aiotter**
## Description of changes
`git filter-repo --help` is broken because man page is not installed.
This PR adds man page.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/git-filter-repo/default.nix [hunk 1]
```diff
@@ -13,10 +15,21 @@ buildPythonPackage rec {
...
+      inherit pname version;
+      hash = "sha256-/hdT4Y8L1tPJtXhoyAEa59BWpuurcGcGOWoV71MScl4=";
+    })
+    (fetchFromGitHub {
+      name = "docs";
+      owner = "newren";
+      repo = pname;
+      rev = "01ead411966a83dfcfb35f9d2e8a9f7f215eaa65";
+      hash = "sha256-Z/3w3Rguo8sfuc/OQ25eFbMfiOHjxQqPY6S32zuvoY4=";
+    })
```
Comment (natsukium) [inline pkgs/development/python-modules/git-filter-repo/default.nix:29]: I think it would be easier to understand declaring a variable named `docs` rather than using srcs.

Comment (natsukium) [reply pkgs/development/python-modules/git-filter-repo/default.nix:29]: Can we add `passthru.updateScripts`?
Automatic package updates would cause inconsistencies between the application and the documentation.
Perhaps you can pass an option like `--version=branch=docs` to nix-update-script.

File: pkgs/development/python-modules/git-filter-repo/default.nix [hunk 2]
```diff
@@ -43,6 +61,6 @@ buildPythonPackage rec {
     description = "Quickly rewrite git repository history";
     homepage = "https://github.com/newren/git-filter-repo";
     license = with licenses; [ mit /* or */ gpl2Plus ];
-    maintainers = with maintainers; [ fab ];
+    maintainers = with maintainers; [ aiotter fab ];
```
Comment (natsukium) [inline pkgs/development/python-modules/git-filter-repo/default.nix:64]: This change should be a separate commit.
Adding the maintainer should be unrelated to adding the man page.

File: pkgs/development/python-modules/git-filter-repo/default.nix (hunk 1 above)
Comment (aiotter) [reply pkgs/development/python-modules/git-filter-repo/default.nix:29]: I found that `nix-update` only updates versions specified as `src` attribute. It's not possible to upgrade `docs` attribute, which is our case.
I'll take a look at other projects and check how they upgrade their sources when they have multiple sources.

**Comment by aiotter** [comment]
It seems to pass all the checks except for missing tests.

@fabaff Can you review my PR as a maintainer of this package?

**Comment by aiotter** [comment]
@natsukium Thank you for reviewing my PR. I made it better!

File: pkgs/development/python-modules/git-filter-repo/default.nix [hunk 3]
```diff
@@ -1,14 +1,18 @@
...
 , pythonOlder
 , setuptools-scm
+, writeScript
 }:
 
 buildPythonPackage rec {
   pname = "git-filter-repo";
   version = "2.38.0";
+  docs_version = "01ead411966a83dfcfb35f9d2e8a9f7f215eaa65";
   format = "setuptools";
```
Comment (natsukium) [inline pkgs/development/python-modules/git-filter-repo/default.nix:16]:
```suggestion
  pyproject = true;
```
Could you please replace `format` with `pyproject` to avoid the legacy build?

> Note that the pyproject format falls back to using setuptools, so you can use pyproject = true even if the package only has a setup.py. 
When unset, the legacy setuptools hooks are used for backwards compatibility.
https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#buildpythonpackage-parameters-buildpythonpackage-parameters

File: pkgs/development/python-modules/git-filter-repo/default.nix [hunk 4]
```diff
@@ -18,6 +22,13 @@ buildPythonPackage rec {
     hash = "sha256-/hdT4Y8L1tPJtXhoyAEa59BWpuurcGcGOWoV71MScl4=";
   };
 
+  docs = fetchFromGitHub {
+    owner = "newren";
+    repo = pname;
```
Comment (natsukium) [inline pkgs/development/python-modules/git-filter-repo/default.nix:27]:
```suggestion
    repo = "git-filter-repo";
```
We should not use `repo = pname` pattern. This is because `repo` and `pname` is semantically different.
see also https://github.com/NixOS/nixpkgs/issues/277994

**Review by fabaff — COMMENTED** [review]
@natsukium pointed out the issues already.

Besides that the change looks solid.

**Comment by aiotter** [comment]
@natsukium @fabaff Thank you for your reviews! I committed the suggestion👍

**Review by natsukium — APPROVED** [review]
LGTM, thanks.

**Comment by natsukium** [comment]
@ofborg eval
